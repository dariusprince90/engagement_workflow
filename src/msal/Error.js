import PropTypes from 'prop-types';

const propTypes = {
  error: PropTypes.object.isRequired
};

const Error = ({ error }) => {
  return (
    <div className="msal-error">
      <p>An authentication error occurred:</p>
      <pre>{JSON.stringify(error, null, 2)}</pre>
    </div>
  );
};

Error.propTypes = propTypes;

export default Error;
