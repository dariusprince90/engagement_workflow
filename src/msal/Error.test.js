import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import faker from '@faker-js/faker';

import Error from './Error';

// **********************************************************************
// * constants

const error = {
  property1: faker.random.alphaNumeric(),
  property2: faker.random.words()
};

const errorForSnapshot = {
  property1: 12345,
  property2: 'This is a mock message for the snapshot tests.'
};

// **********************************************************************
// * mock external dependencies

// **********************************************************************
// * unit tests

describe('AppLoader', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Error error={error} />, div);
  });

  it('renders correct layout', () => {
    const tree = renderer.create(<Error error={errorForSnapshot} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
