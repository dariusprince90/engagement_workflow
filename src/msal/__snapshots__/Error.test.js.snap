// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLoader renders correct layout 1`] = `
<div
  className="msal-error"
>
  <p>
    An authentication error occurred:
  </p>
  <pre>
    {
  "property1": 12345,
  "property2": "This is a mock message for the snapshot tests."
}
  </pre>
</div>
`;
